experimental/Geometry/Schemes.jl

```diff
@@ -0,0 +1,1244 @@
+  ϕ = AlgebraHomomorphism( R, R, gens(R) )
+  f = AffSchMorphism(D,D,ϕ)
+  return f
+end
+
+#####################################################################
+# Composition of affine scheme morphisms. 
+# Consists basically of composition of the associated pullback 
+# homomorphisms.
+#
+function compose( f::AffSchMorphism, g::AffSchMorphism ) 
+  if codomain(f) != domain(f) 
+    error( "morphisms can not be composed" )
+  end
+  ϕ = pullback(f)
+  γ = pullback(g)
+  return AffSchMorphism( domain(g), codomain(f), compose( γ, ϕ ) )
+end
+
+#####################################################################
+# Glueing of two affine schemes X and Y along an open subset 
+# U ⊂ X and V ⊂ Y via an isomorphism Φ : U → V. 
+# A glueing consists of a diagram X ↩ U → V ↪ Y with the outer maps 
+# the open embeddings of principal open subsets. 
+#
+mutable struct Glueing
+  firstPatch::AffineScheme
+  secondPatch::AffineScheme
+  inclusionFirstPatch::AffSchMorphism
+  inclusionSecondPatch::AffSchMorphism
+  glueingIsomorphism::AffSchMorphism
+
+  function Glueing( 
+      inclusionFirstPatch, 
+      inclusionSecondPatch, 
+      glueingIsomorphism 
+    )
+    if domain( inclusionFirstPatch ) != domain( glueingIsomorphism ) 
+      error( "Morphisms can not be composed for glueing." )
+    end
+    if codomain( glueingIsomorphism ) != domain( inclusionSecondPatch ) 
+      error( "Morphisms can not be composed for glueing." )
+    end
+    Γ = new()
+    Γ.firstPatch = codomain( inclusionFirstPatch )
+    Γ.secondPatch = codomain( inclusionSecondPatch )
+    Γ.inclusionFirstPatch = inclusionFirstPatch
+    Γ.inclusionSecondPatch = inclusionSecondPatch
+    Γ.glueingIsomorphism = glueingIsomorphism
+    return Γ
+  end
+end
+
+first_patch( Γ::Glueing ) = Γ.firstPatch
+second_patch( Γ::Glueing ) = Γ.secondPatch
+overlap(Γ::Glueing) = domain(Γ.inclusionFirstPatch)
+first_inclusion( Γ::Glueing ) = Γ.inclusionFirstPatch
+second_inclusion( Γ::Glueing ) = Γ.inclusionSecondPatch
+glueing_isomorphism( Γ::Glueing ) = Γ.glueingIsomorphism
+
+function Base.show( io::Base.IO, Γ::Glueing )
+  Base.print( io, "Glueing of\n" )
+  Base.print( io, Γ.firstPatch )
+  Base.print( io, "\nand\n" )
+  Base.print( io, Γ.secondPatch )
+  Base.print( "\nalong\n" )
+  Base.print( io, domain(Γ.inclusionFirstPatch) )
+  return
+end
+
+################################################################################
+# Covering of a scheme.
+# It consists of a list of affine patches together with their glueings.
+# The glueings are stored in an array (which is usually half empty, since 
+# glueings need only be provided for ordered pairs of indices). 
+# The get_index methods for double arrays have been overloaded so that 
+# OrderedMultiindex can be used to address them.
+#
+mutable struct Covering
+  patches::Vector{AffineScheme}	# A list of open patches for this scheme
+  glueings::Array{Union{Glueing,Nothing},2}     # A list of glueings between the patches 
```

Comment:
```suggestion
  glueings::Matrix{Union{Glueing,Nothing}}     # A list of glueings between the patches 
```
... and similar in many other places